recipes/console_bridge/all/patches/disable-pkgconfig-export.patch

```diff
@@ -0,0 +1,20 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -72,6 +72,8 @@
+ install(FILES ${PROJECT_BINARY_DIR}/${PROJECT_NAME}_export.h
+   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+ 
++if(FALSE)
```

Comment:
Do you mean the hook error? Let's take a look carefully in the message then:

```
[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown folder 'CMake' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013-DEFAULT-PACKAGE-LAYOUT) 
[HOOK - conan-center.py] post_package(): ERROR: [CMAKE-MODULES-CONFIG-FILES (KB-H016)] The conan-center repository doesn't allow the packages to contain CMake find modules or config files. The packages have to be located using generators and the declared `cpp_info` information (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H016-CMAKE-MODULES-CONFIG-FILES) 
[HOOK - conan-center.py] post_package(): ERROR: [CMAKE-MODULES-CONFIG-FILES (KB-H016)] Found files: ./CMake/console_bridge-config.cmake; ./CMake/console_bridge-config.cmake (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H016-CMAKE-MODULES-CONFIG-FILES) 
```

The hook says there is a folder named `CMake` in package folder. That's not rare, and usually happens when building on Windows. Few lines above we have:

```
-- Installing: C:/J/w/prod/BuildSingleReference/.conan/data/console_bridge/1.0.2/_/_/package/127af201a4cdf8111e2e08540525c245c9b3b99e/CMake/console_bridge-targets.cmake
```

So to fix it, you just need to add:

```python
rmdir(self, os.path.join(self.package_folder, "CMake"))
```

And you don't need to used an `if` because Conan will not complain in case the folder does not exist. I hope it fixes your error.

Reply:
Thanks. I already tried it here: https://github.com/conan-io/conan-center-index/compare/88650fac29272d78f92bd195d1218518bfda5335..f625ca6ea55410dfce3e3fbdf3299f2d24e971d8
and it failed with this error: https://github.com/conan-io/conan-center-index/pull/17346#issuecomment-1591281141